### PR TITLE
Skip dependency `Gemfile.lock` during the artifact creation.

### LINKED ISSUE #14877: skip dependency "Gemfile.lock" files during artifact creation
Currently artifacts will carry Gemfile.lock files from any gem dependency that decides to not exclude it from the .gem file.

Example:
```
/tmp/logstash-8.6.0 ❯ find vendor -name Gemfile.lock
./vendor/bundle/jruby/2.6.0/gems/http_parser.rb-0.6.0-java/Gemfile.lock
./vendor/bundle/jruby/2.6.0/gems/pleaserun-0.0.32/Gemfile.lock
./vendor/bundle/jruby/2.6.0/gems/amazing_print-1.4.0/Gemfile.lock
./vendor/bundle/jruby/2.6.0/gems/insist-1.0.0/Gemfile.lock
```

This can confuse dependency vulnerability scanners that pick any "Gemfile.lock" to determine dependencies and their versions.

### PR DESCRIPTION
**Pull request by mashhurs**
## Release notes

[rn:skip] 

## What does this PR do?

Excludes dependency Gemfile.lock files (example `./vendor/bundle/jruby/2.6.0/gems/http_parser.rb-0.6.0-java/Gemfile.lock`) during the artifact creation. This will help vulnerability scanners not to pick up while scanning.

## Why is it important/What is the impact to the user?
N.A

## Checklist

- [x] My code follows the style guidelines of this project
- ~[ ] I have commented my code, particularly in hard-to-understand areas~
- ~[ ] I have made corresponding changes to the documentation~
- ~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~

## Author's Checklist
- [ ]

## How to test this PR locally
- Clone current change and create any artifact (example: `./gradlew assembleTarDistribution`)
- Check that if created artifact contains vendor Gemfile.lock files (`find vendor -name Gemfile.lock`).

## Related issues

- Closes #14877

### CODE REVIEW

**Comment by mashhurs** [comment]
Failed CIs are due to 8.8.0 artifact not found.
```
```

File: rakelib/artifacts.rake [hunk 1]
```diff
@@ -95,6 +95,9 @@ namespace "artifact" do
     @exclude_paths << 'vendor/**/gems/*/test/**/*'
     @exclude_paths << 'vendor/**/gems/*/spec/**/*'
 
+    # dependency vulnerability scanners may pick unnecessary Gemfile.lock files
+    @exclude_paths << 'vendor/**/gems/*/Gemfile.lock'
```
Comment (jsvd) [inline rakelib/artifacts.rake:99]: Let's also exclude 'Gemfile', as they're ignored (more on this here: https://bundler.io/guides/faq.html#using-gemfiles-inside-gems)

```suggestion
    @exclude_paths << 'vendor/**/gems/*/Gemfile.lock'
    @exclude_paths << 'vendor/**/gems/*/Gemfile'
```

File: rakelib/artifacts.rake [hunk 2]
```diff
@@ -95,6 +95,9 @@ namespace "artifact" do
     @exclude_paths << 'vendor/**/gems/*/test/**/*'
     @exclude_paths << 'vendor/**/gems/*/spec/**/*'
 
+    # dependency vulnerability scanners may pick unnecessary Gemfile.lock files
```
Comment (mashhurs) [inline rakelib/artifacts.rake:98]:
```suggestion
    # vulnerability scanners shouldn't pick dependency Gemfile
```

File: rakelib/artifacts.rake (hunk 1 above)
Comment (mashhurs) [reply rakelib/artifacts.rake:99]: Good suggestion, thanks.
It seems we should also exclude the shared gems (ex: `vendor/jruby/lib/ruby/gems/shared/gems/net-telnet-0.1.1/Gemfile`) so the patter would be:
```
    @exclude_paths << 'vendor/**/gems/**/Gemfile.lock'
    @exclude_paths << 'vendor/**/gems/**/Gemfile'
```

Comment (mashhurs) [inline rakelib/artifacts.rake:99]:
```suggestion
    @exclude_paths << 'vendor/**/gems/**/Gemfile.lock'
    @exclude_paths << 'vendor/**/gems/**/Gemfile'
```

**Review by jsvd — APPROVED** [review]
LGTM

**Pull request merged**